src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.mux.cs

```diff
@@ -26,7 +26,7 @@ namespace Microsoft/* UWP don't rename */.UI.Xaml.Controls;
 
 partial class RatingControl
 {
-#if !HAS_UNO // TODO Uno: Unused on Uno currently
+#if __SKIA__ // TODO Uno: Expression animation is only supported on Skia
```

Comment:
@Youssef1313 shouldn't the `ExpressionAnimation` type be marked with `[NotImplemented]` for the targets where it does not actually have effect?

https://github.com/unoplatform/uno/blob/56f15f541bc03bdc6b07e8e8be3baf2406f51aff/src/Uno.UI.Composition/Composition/ExpressionAnimation.cs#L7

Reply:
`ExpressionAnimation`, on its own, doesn't have any platform-specifics. I think it's `CompositionObject.StartAnimation` method that's really not implemented on the other platforms.

Still, the only use of `ExpressionAnimation` is really with `StartAnimation`, so it's unusable if `StartAnimation` doesn't work on the other platforms.

Similar thing with InteractionTracker. I'll open an issue to track this.

Reply:
The reason I was asking is that originally I made the code with `ApiInformation.IsTypeAvailable` check for `ExpressionAnimation` and it used that code path which surprised me 😀 . The `StartAnimation` check is ok, but it could be useful if we were able to see the `NotImplemented` analyzer underlines _while_ working on Uno.UI itself - as then we would have easier time noticing when a given part of code is not implemented and we would be able to adjust more easily.

Reply:
The analyzer is currently not enabled in Uno.UI, unfortunately. There will be lots of warnings, but we could try to take a step and enabling it.